@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FormEventHandler } from 'react'
+import { Map } from 'immutable'
+import { withState } from 'reaclette'
+
+import Button from './Button'
+import Input from './Input'
+import IntlMessage from './IntlMessage'
+import Select, { Options } from './Select'
+import { alert } from './Modal'
+
+import XapiConnection, { ObjectsByType, Pif, PifMetrics } from '../libs/xapi'
+
+interface ParentState {
+  objectsByType: ObjectsByType
+  objectsFetched: boolean
+  xapi: XapiConnection
+}
+
+interface State {
+  formRef: React.RefObject<HTMLFormElement>
+  isBonded: boolean
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  _newNetwork: FormEventHandler<HTMLFormElement>
+  _resetForm: () => void
+  _toggleBonded: () => void
+}
+
+interface Computed {
+  pifs?: Map<string, Pif>
+  pifsMetrics?: Map<string, PifMetrics>
+}
+
+const OPTIONS_RENDER_PIF: Options<Pif> = {
+  render: (pif, additionalProps) =>
+    `${pif.device} (${
+      additionalProps?.pifsMetrics?.find((metrics: PifMetrics) => metrics.$ref === pif.metrics)?.device_name ??
+      'unknown'
+    })`,
+  value: pif => pif.$id,
+}
+const OPTIONS_RENDER_BOND_MODE: Options<string[]> = {
+  render: mode => mode,
+  value: mode => mode,
+}
+const BOND_MODE = ['balance-slb', 'active-backup', 'lacp']
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      formRef: React.createRef(),
+      isBonded: false,
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+    },
+    effects: {
+      _newNetwork: async function (e) {
+        e.preventDefault()
+        const { current } = this.state.formRef
+        const bondMode = current?.bondMode?.value
+        const desc: string = current?.desc.value || 'Created with Xen Orchestra Lite'
+        const mtu = +current?.mtu.value || 1500
+        const name: string | undefined = current?.networkName.value
+        const pifsId: string | string[] | undefined = this.state.isBonded
+          ? Array.from<HTMLOptionElement>(current?.pif.selectedOptions).map(({ value }) => value)
+          : current?.pif.value
+        const vlan = +current?.vlan?.value || 0
+
+        let networkRef: string | undefined
+        try {
+          networkRef = (await this.state.xapi.call('network.create', {
+            name_label: name,
+            name_description: desc,
+            other_config: { automatic: 'false' },
+            MTU: mtu,
+            VLAN: vlan,
+          })) as string
+
+          if (this.state.isBonded && Array.isArray(pifsId)) {
+            const pifsRefList = pifsId.map(pifId => this.state.pifs?.get(pifId)?.$network.PIFs)
+            await Promise.all(
+              pifsRefList.map(pifs => this.state.xapi.call('Bond.create', networkRef, pifs, '', bondMode))
+            )
+            this.effects._toggleBonded()
```

Comment:
Why? `_resetForm()` should be enough.

Reply:
`resetForm` don't have effect on the `isBonded` input, because it's a `controlled field`.